src/components/drop/Drop.tsx

```diff
@@ -52,17 +52,20 @@ export class Drop extends React.PureComponent<IDropProps> {
 
         this.button = React.createRef();
         this.handleDocumentClick = this.handleDocumentClick.bind(this);
-    }
-
-    componentWillMount() {
-        document.addEventListener('mousedown', this.handleDocumentClick);
+        this.onClick = this.onClick.bind(this);
     }
 
     componentWillUnmount() {
-        document.removeEventListener('mousedown', this.handleDocumentClick);
+        this.removeEventOnClickOnDocument();
     }
 
     render() {
+        if (this.props.isOpen) {
+            this.setEventOnClickOnDocument();
+        } else {
+            this.removeEventOnClickOnDocument();
+        }

```

Comment:
This if block should be done inside `componentDidUpdate`, because you will be setting a new event handler on each render if `this.props.isOpen` remains true.